staging/src/k8s.io/apimachinery/pkg/runtime/error.go

```diff
@@ -19,53 +19,76 @@ package runtime
 import (
 	"fmt"
 	"reflect"
+	"strconv"
 	"strings"
 
 	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
 type notRegisteredErr struct {
-	schemeName string
-	gvk        schema.GroupVersionKind
-	target     GroupVersioner
-	t          reflect.Type
+	context fmt.Stringer
+	gvk     schema.GroupVersionKind
+	target  GroupVersioner
+	t       reflect.Type
 }
 
 func NewNotRegisteredErrForKind(schemeName string, gvk schema.GroupVersionKind) error {
-	return &notRegisteredErr{schemeName: schemeName, gvk: gvk}
+	return &notRegisteredErr{context: schemeContext(schemeName), gvk: gvk}
 }
 
 func NewNotRegisteredErrForType(schemeName string, t reflect.Type) error {
-	return &notRegisteredErr{schemeName: schemeName, t: t}
+	return &notRegisteredErr{context: schemeContext(schemeName), t: t}
 }
 
 func NewNotRegisteredErrForTarget(schemeName string, t reflect.Type, target GroupVersioner) error {
-	return &notRegisteredErr{schemeName: schemeName, t: t, target: target}
+	return &notRegisteredErr{context: schemeContext(schemeName), t: t, target: target}
 }
 
 func NewNotRegisteredGVKErrForTarget(schemeName string, gvk schema.GroupVersionKind, target GroupVersioner) error {
-	return &notRegisteredErr{schemeName: schemeName, gvk: gvk, target: target}
+	return &notRegisteredErr{context: schemeContext(schemeName), gvk: gvk, target: target}
+}
+
+func NewNotRegisteredGVKErrForTargetWithContext(context string, gvk schema.GroupVersionKind, target GroupVersioner) error {
+	return &notRegisteredErr{context: genericContext(context), gvk: gvk, target: target}
+}
+
+type schemeContext string
+
+func (s schemeContext) String() string {
+	if len(s) == 0 {
+		return ""
+	}
+	return " in scheme " + strconv.Quote(string(s))
```

Comment:
nit: Padding the context with a space prefix here feels clunky.  Could we do all the formatting in one place?

Reply:
@jpbetz the reason for this is in how the `notRegisteredErr` string is formatted here: https://github.com/MadhavJivrajani/kubernetes/blob/fa6dcc734bf6d6faa720714628b8709c27736f19/staging/src/k8s.io/apimachinery/pkg/runtime/error.go#L73

There are two main cases to take care of: when we use a `genericContext` and when we use a `schemeContext`. If the `genericContext` is used (meaning we already know what the scheme is, such as in cases of built in types), things would be okay for the most part, however, if we use a `schemeContext` (for when we don't know the scheme before hand such as in CRs), we need to add in a ` in scheme ` there towards the _end_ of the formatted `notRegisteredErr` error string. So to handle both cases and not overcomplicate the formatting logic, a space is padded here.